game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java

```diff
@@ -601,6 +601,7 @@ public void fight(final IDelegateBridge bridge) {
     }
     bridge.getHistoryWriter().startEvent("Battle in " + battleSite, battleSite);
     removeAirNoLongerInTerritory();
+    markAttackingTransports(bridge);
```

Comment:
This could potentially be slightly simpler written as:
```
    if (!headless) {
      markAttackingTransports(bridge);
      writeUnitsToHistory(bridge);
    }
```

Essentially we're bringing the no-op method check to the point of the caller instead of in the method. This way if you are looking at the higher level flow, you can more easily follow the branching logic. As-is, with a series of void methods, to some extent they almost may as well be inlined with demarcating comments, a reader has to jump through each one to understand what is happening. Hence, bringing the logic flow out of the lower level methods allows a reader to see the code flow at a higher level without jumping to each method.